File: src/main/java/gregtechlite/gtlitecore/mixins/MixinExtension.java
package gregtechlite.gtlitecore.mixins;

import java.lang.annotation.ElementType;
import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;
import java.lang.annotation.Target;

/**
 * Extension attributor for all mixins classes.
 * <p>
 * This annotation used to mark an interface is an extension for mixins class.
 *
 * @see MixinOnly
 */
@Target(ElementType.TYPE)
@Retention(RetentionPolicy.SOURCE)
public @interface MixinExtension {}

File: src/main/kotlin/gregtechlite/gtlitecore/api/mixins/MixinExtension.kt
package gregtechlite.gtlitecore.api.mixins

/**
 * Marked an interface is extension for mixins class or only used for mixins class (means it as the hook for the
 * specific mixins class).
 *
 * @see gregtechlite.gtlitecore.mixins
 */
@Target(AnnotationTarget.CLASS)
@Retention(AnnotationRetention.SOURCE)
annotation class MixinExtension()
